internal/pkg/siderolink/manager.go

```diff
@@ -751,6 +755,16 @@ func (manager *Manager) Provision(ctx context.Context, req *pb.ProvisionRequest)
 		dirty = true
 	}
 
+	if spec.RemoteAddr != host {
+		if _, err = safe.StateUpdateWithConflicts(ctx, manager.state, link.Metadata(), func(r *siderolink.Link) error {
+			r.TypedSpec().Value.RemoteAddr = host
```

Comment:
dumb question - why do even collect that?

Reply:
That's a long story, but it might help in case when you have a machine joining your account and Omni can't get access to it. And you can't even tell which node is that.